plugin/core/views.py

```diff
@@ -172,3 +178,42 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
         "options": formatting_options(view.settings()),
         "range": region_to_range(view, region).to_lsp()
     })
+
+
+def minihtml(view: sublime.View, content: Union[str, dict], popup_width: Optional[int] = None) -> str:
+    """ Content can be a string or MarkupContent. `popup_width` will wrap text based on the provided width. """
+    if isinstance(content, str):
+        return text2html(view, content, popup_width)
+    elif isinstance(content, dict):
+        value = content.get("value") or ""
+        kind = content.get("kind")
+        if kind == "markdown":
+            return mdpopups.md2html(view, value)
+        else:
+            # must be plaintext
+            return text2html(view, value, popup_width)
+    else:
+        return ''
+
+
+def text2html(view: sublime.View, content: str, popup_width: Optional[int] = None) -> str:
+    content = html.escape(content).replace('\n', '<br>').replace('\t', '&nbsp;&nbsp;&nbsp;&nbsp;')
+    if popup_width is not None:
+        # if popup width is provided, calucalte how many caracters can fit on one line
+        wrap_width = int(popup_width / view.em_width())
+        content = "<br>".join(textwrap.wrap(content, width=wrap_width,
+                              replace_whitespace=False, break_long_words=False, break_on_hyphens=False))
+
+    def replace_nbsp(match: Any) -> str:
+        spaces = match.group(0)
+        return "&nbsp;" * len(spaces)
+
+    # if there are 2 or more spaces, replace them with &nbsp;
+    content = re.sub(r"( {2,})", replace_nbsp, content)
+
+    def replace_url_with_link(match: Any) -> str:
+        url = match.group(0)
+        return "<a href='{}'>{}</a>".format(url, url)
+
+    FIND_URL = re.compile(r'(https?://(?:[\w\d:#@%/;$()~_?\+\-=\\\.&](?:#!)?)*)', flags=re.IGNORECASE)
```

Comment:
Sorry, one more thing:

The link is not "linkified" properly and clicking it opens blank tab:
![Screenshot 2020-05-12 at 21 54 51](https://user-images.githubusercontent.com/153197/81739312-672c0000-949b-11ea-89ec-99a8cfb723a8.png)

Reply:
Before and after parsing link:
```
Extensible JSON &lt;http://json.org&gt; encoder
=>
Extensible JSON &lt;<a href='http://json.org&gt;'>http://json.org&gt;</a> encoder
```
So the escaping is what causes the issue.
Can we replace links before escaping?

Reply:
No, we can't because we'll escape html tags added by us...

Reply:
I can probably detect links before encoding 
```
detected_links = []

# before encoding

text = "Extensible JSON <http://json.org>; encoder"
#                ^              ^ link detected

detected_links = ['http://json.org']

# after encoding

text = "Extensible JSON &lt;http://json.org&gt; encoder"

for url in detected_links:
  text.replace(url, "<a href='{}'>{}</a>".format(url, url))

Reply:
Also the "table" is pretty misaligned in that same popup:
![Screenshot 2020-05-12 at 22 34 59](https://user-images.githubusercontent.com/153197/81742766-e7a12f80-94a0-11ea-9655-f9a011d87ff6.png)

Reply:
The "table" is the reason on why I think I should remove the text wrap.
Although I don't like vertical scroll (and textwrap seemed to fix it), I would rather see a scroll than a broken "table" .